[BridgeXPathAbstract + BlizzardNewsBridge + XPathBridge] Add new abstract class + two example implementations

Niehztog opened the pull request:
As a reaction to the discussion in my [original pull request](https://github.com/RSS-Bridge/rss-bridge/pull/1645), where user @somini suggested to provide an abstract bridge simplifying the creation of new bridges using xpath expressions, I developed this new proposal as a pull request.
Basically I divided the code of my [original pull request](https://github.com/RSS-Bridge/rss-bridge/pull/1645) into three different classes: One abstract class "BridgeXPathAbstract" and two derived classes "BlizzardNewsBridge" and "XPathBridge" demonstrating how to use it. This approach has two great advantages:
- other bridge developers can make use of the xpath functions more easily, shortening time required to develop and update bridges
- user defined xpath expressions for parsing new sites into feeds can easily be stored in a new bridge file which can be shared with the whole community

**Contents:**
- **BridgeXPathAbstract** is a new base class for making development of new bridges more easy.
- **BlizzardNewsBridge** is an example implementation which covers news feed of game company blizzard in multiple languages.
- **XPathBridge** is an example implementation which allows entering any user defined xpath expressions into form input fields (great flexibility).

This PR can be regarded as an alternative approach to [1645](https://github.com/RSS-Bridge/rss-bridge/pull/1645).
If desired I would be totaly okay with removing XPathBridge from the PR. I left it here for quick debugging of new xpath expressions without the need to create a new bridge class.

Niehztog committed:
new abstract class simplifying creation of new bridges + two exmaple implementations (BlizzardNewsBridge + XPathBridge)

Niehztog committed:
new abstract class simplifying creation of new bridges + two exmaple implementations (BlizzardNewsBridge + XPathBridge)

Niehztog committed:
new abstract class simplifying creation of new bridges + two exmaple implementations (BlizzardNewsBridge + XPathBridge)

Niehztog committed:
new abstract class simplifying creation of new bridges + two exmaple implementations (BlizzardNewsBridge + XPathBridge)

em92 commented on bridges/BlizzardNewsBridge.php:
```diff
@@ -0,0 +1,62 @@
+<?php
+require_once __DIR__ . DIRECTORY_SEPARATOR . '..' . DIRECTORY_SEPARATOR . 'lib' . DIRECTORY_SEPARATOR
+	. 'BridgeXPathAbstract.php';
```
Add `BridgeXPathAbstract.php` importing to `lib/rssbridge.php`

em92 commented on lib/BridgeXPathAbstract.php:
```diff
@@ -0,0 +1,565 @@
+	 * Should provide the feeds title
+	 *
+	 * @param DOMXPath $xpath
+	 * @return string|string[]
```
Not familiar with doc's syntax. Does it include situation, where function does not return anything?

Niehztog replied:
> 
> 
> Not familiar with doc's syntax. Does it include situation, where function does not return anything?

I think it was added in error. I removed this "doc's syntax"

em92 commented on lib/BridgeXPathAbstract.php:
```diff
@@ -0,0 +1,565 @@
+		} elseif(is_string($typedResult) && strlen($typedResult) > 0) {
+			return trim($typedResult);
+		}
+	}
```
It should returnServerError if not string and not DomNodeList

em92 commented on lib/BridgeXPathAbstract.php:
```diff
@@ -0,0 +1,565 @@
+<?php
+
+abstract class BridgeXPathAbstract extends BridgeAbstract {
```
Why not naming as XPathAbstract?

Niehztog replied:
I renamed it. Not sure though what naming choice supports clearity and understanding of its purpose mostly.

em92 reviewed (changes requested):
Nice work, @Niehztog !
Here are some comments. I will add more later

em92 commented:
Also, you don't need to rebase and force push. You commits will be squashed after merging.

Niehztog committed:
various little improvements, incorporating em92's code review feedback

Niehztog committed:
various little improvements, incorporating em92's code review feedback

Niehztog committed:
various little improvements, incorporating em92's code review feedback

Niehztog committed:
various little improvements, incorporating em92's code review feedback

Niehztog commented:
Hello @em92 
thank you very much for your kind and helpful feedback and your explanation about rebasing/force-pushing. I added a second commit implementing most of your suggestions. If you have more, they'd be welcome.

em92 commented:
Hi, @Niehztog ! Could you please rebase your PR? In mid october I have fixed Travis CI for automatic PR checks. It stopped working in August.

Niehztog commented:
> 
> 
> Hi, @Niehztog ! Could you please rebase your PR? In mid october I have fixed Travis CI for automatic PR checks. It stopped working in August.

Hi @em92 , thank your for looking into here! I just rebased this PR. Hopefully it looks fine now.

em92 commented:
gj @Niehztog

dvikan commented on lib/BridgeXPathAbstract.php:
```diff
@@ -0,0 +1,565 @@
+	 */
+	protected function fixEncoding($input)
+	{
+		return $this->getParam('fix_encoding') ? utf8_decode($input) : $input;
```
I think this should be `utf8_encode()` which would convert iso-8859 to utf8. Right now it's the other way around.
